src/components/common/ErrorBoundary.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+interface FallbackProps {
+  error?: Error;
+  resetError: () => void;
+  customMessage?: string;
+}
+
+const ErrorContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+  padding: 20px;
+  text-align: center;
+  background-color: #f8f9fa;
+`;
+
+const ErrorTitle = styled.h1`
+  font-size: 24px;
+  color: #ff6b6b;
+  margin-bottom: 16px;
+`;
+
+const ErrorMessage = styled.p`
+  font-size: 16px;
+  color: #666;
+  margin-bottom: 20px;
+  line-height: 1.5;
+`;
+
+const RetryButton = styled.button`
+  padding: 12px 24px;
+  background-color: #4a90e2;
+  color: white;
+  border: none;
+  border-radius: 8px;
+  font-size: 16px;
+  cursor: pointer;
+  transition: background-color 0.2s;
+
+  &:hover {
+    background-color: #357abd;
+  }
+`;
+
+interface ErrorBoundaryState {
+  hasError: boolean;
+  error?: Error;
+}
+
+interface ErrorBoundaryProps {
+  children: React.ReactNode;
+  fallback?: React.ComponentType<FallbackProps>;
+  customMessage?: string; // 커스텀 메시지 prop 추가
+}
+
+export class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError(error: Error): ErrorBoundaryState {
+    return { hasError: true, error };
+  }
+
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
+    console.error('ErrorBoundary caught an error:', error, errorInfo);
+  }
+
+  resetError = () => {
+    this.setState({ hasError: false, error: undefined });
+  };
+
+  render() {
+    if (this.state.hasError) {
+      if (this.props.fallback) {
+        const FallbackComponent = this.props.fallback;
+        return (
+          <FallbackComponent
+            error={this.state.error}
+            resetError={this.resetError}
+            customMessage={this.props.customMessage}
+          />
+        );
```

Comment:
[권장]
`fallback`은 공통 에러 UI 대신 별도의 UI를 사용하기 위해 만든 props 같은데요. `fallback`을 넘겨줄때 `customMessage`까지 같이 props로 넘겨줘야 하는게 ErrorBoundary를 사용하는 입장에선 다음과 같이 헷갈릴 수 있을 것 같아요.

- "fallback을 제공했는데 customMessage도 같이 써야 하나?"
- "fallback만 쓰면 customMessage는 무시되나?"
- "customMessage만 사용하려면 fallback은 undefined로 두어야 하나?"

`fallback`이 있다면 사용하는 쪽에서 완전히 UI를 구성하도록 위임하는 식으로 `fallback`과 `customMessage`의 역할을 명확히 분리하여 개발자 입장에서의 사용 경험을 개선해보시는 것을 권장드립니다.

예시)

```suggestion
        const FallbackComponent = this.props.fallback;
        return (
          <FallbackComponent
            error={this.state.error}
            resetError={this.resetError}
          />
        );
```